resources/sip11/ra/src/test/java/org/mobicents/slee/resource/sip11/test/SipResourceAdaptorTest.java

```diff
@@ -0,0 +1,144 @@
+package org.mobicents.slee.resource.sip11.test;
+
+import javax.sip.ListeningPoint;
+import javax.slee.facilities.Tracer;
+import javax.slee.resource.ConfigProperties;
+import javax.slee.resource.ConfigProperties.Property;
+import javax.slee.resource.ResourceAdaptorContext;
+
+import org.mobicents.ha.javax.sip.ClusteredSipStack;
+import org.mobicents.slee.resource.sip11.SipResourceAdaptor;
+import org.mobicents.slee.resource.sip11.SleeSipProviderImpl;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.BeforeClass;
+
+import org.powermock.reflect.Whitebox;
+
+import static org.mockito.Matchers.*;
+import static org.mockito.Mockito.*;
+import org.mockito.stubbing.*;
+import org.mockito.invocation.*;
+import java.util.Arrays;
+import java.util.Iterator;
+
+public class SipResourceAdaptorTest {
+
+    private  static final int SIP_RA1_PORT = 5060;
+	private static final int SIP_RA2_PORT = 5059;
+	private static final String STACK_ADDRESS = "127.0.0.1";
+    // common mocked tracer
+	private static Tracer tracer1 = mock(Tracer.class);
+
+	// resource adaptor context
+	private static ResourceAdaptorContext raContext1 = mock(ResourceAdaptorContext.class);
+	private static ResourceAdaptorContext raContext2 = mock(ResourceAdaptorContext.class);
+
+	// slee sip provider mock
+	private static SleeSipProviderImpl sleeSipProvider = mock(SleeSipProviderImpl.class);
+
+	// minimal mock just to satisfy SipResourceAdaptor's usage of the tracer
+	private static void mockTracer() {
+		// mock info() call for tracer
+		Answer<Object> answer = new Answer<Object>() {
+			public Object answer(InvocationOnMock invocation) {
+				System.out.println(" (TRACER-MOCK) :: info() called :: " + Arrays.toString(invocation.getArguments()));
+				return null;
+			}
+		};
+		doAnswer(answer).when(tracer1).info(anyString());
+	}
+
+	// minimal mock just to satisfy SipResourceAdaptor's usage of the tracer
+	private static void mockRaContext() {
+		when(raContext1.getEntityName()).thenReturn("stack#1");
+		when(raContext2.getEntityName()).thenReturn("stack#2");
+	}
+
+	@BeforeClass
+	public static void prepareEnvSharedByAllTests() {
+		mockTracer();
+		mockRaContext();
+	}
+
+	@Test
+	/*
+	 * Purpose of this test it to verify whether two RA entities can be activated on the
+	 * same network interface within same class loader.
+	 */
+	public void testActivateTwoRaEntities() {
+		// create two RAs
+		SipResourceAdaptor ra1 = new SipResourceAdaptor();
+		SipResourceAdaptor ra2 = new SipResourceAdaptor();
+
+		// initialize internal dependencies of RA#1
+		Whitebox.setInternalState(ra1, "tracer", tracer1);
+		Whitebox.setInternalState(ra1, "raContext", raContext1);
```

Comment:
Yes, can be used, but still raContext mock should deliver some predictable tracer mock.